tee-worker/litentry/core/assertion-build/src/a10.rs

```diff
@@ -53,13 +49,20 @@ pub fn build(req: &AssertionBuildRequest, min_balance: ParameterString) -> Resul
 
 	let mut is_hold = false;
 	let mut optimal_hold_index = 0;
-	for index in 0..ASSERTION_FROM_DATE.len() {
+	for (index, date) in ASSERTION_FROM_DATE.iter().enumerate() {
 		if is_hold {
 			break
 		}
 
 		for address in &addresses {
-			match client.is_holder(&req.assertion, address, index) {
+			let holding = ParamsBasicTypeWithAmountHoding::new(
+				"ethereum".into(),
```

Comment:
What do you think about using web3network here and mapping it into achainable chain ? With that we will reduce the amount of these consts spread across the code.